types/emailaddress.pp

```diff
@@ -0,0 +1,2 @@
+# naive email address validation
+type Tea::EmailAddress = Pattern[/\A(?:[\w+\-].?)+@[a-z\d\-]+(?:\.[a-z]+)*\.[a-z]+\z/]
```

Comment:
Domains can also have numeric chars. Some tools might also accept IDN domains rather than the punycode. This doesn't allow that.

Reply:
`.+@.+` is maybe better than `.*@.*`?  I do a fair bit of interpolation in hiera.  So even *very* simple validation often catches errors for me.

eg This contrieved example where I've made a typo in `fqdn`.
`foo::email_address: "alex.fisher@%{::fqdm}"`

Reply:
That's what I had in mind. Mornings and regexes aren't a good idea.